actionpack/lib/action_controller/test_case.rb

```diff
@@ -208,10 +207,14 @@ def load!
   end
 
   # Superclass for ActionController functional tests. Functional tests allow you to
-  # test a single controller action per test method. This should not be confused with
-  # integration tests (see ActionDispatch::IntegrationTest), which are more like
-  # "stories" that can involve multiple controllers and multiple actions (i.e. multiple
-  # different HTTP requests).
+  # test a single controller action per test method.
+  #
+  # == Use `ActionDispatch::Integration` tests instead of this.
+  #
+  # Rails discourages use of `ActionController` functional tests in favor of integration tests
+  # (see ActionDispatch::IntegrationTest).
+  # New Rails applications no longer generate `ActionController` style functional tests and they should
+  # only be used for backward compatibility. Use `ActionDispatch::IntegrationTest` instead.
```

Comment:
Learned yesterday that our documentation doesn't handle backticks too well. So we'd have to go for `+` or `<tt></tt>`

See the section about request encoding here: http://api.rubyonrails.org/v5.0.0.1/classes/ActionDispatch/IntegrationTest.html (I've already fixed this in edge).